File: src/document.cpp
#include "document.h"

namespace Bypass {

	Document::Document()
	: elements()
	{

	}

	Document::~Document() {

	}

	void Document::append(const Element& element) {
		elements.push_back(Element(element));
	}

	size_t Document::size() {
		return elements.size();
	}

	Element Document::operator[](size_t i) {
		return elements[i];
	}

}

File: platform/android/src/in/uncod/android/bypass/Document.java
package in.uncod.android.bypass;

public class Document {

	List<Element> elements;
	
	public Document(List<Element> elements) {
		this.elements = elements;
	}
	
	public List<Element> getElements() {
		return elements;
	}
	
	public int getElementCount() {
		return elements.size();
	}
	
	public Element getElement(int pos) {
		return elements.get(pos);
	}
}
